src/fmu/tools/rms/qcreset.py

```diff
@@ -209,3 +209,179 @@ def set_surfaces(
+            RMS project.
+
+        horizons: A Python dictionary where each key corresponds to the name of
+            the horizons category where horizon data need to be made empty. The
+            value associated to this key should be a list of horizon names to
+            modify. If a string ``all`` is assigned instead of a list, all
+            available horizon names for this category will be used.
+            Alternatively, if a list of horizons categories is given instead of
+            a dictionary, the method will apply to all horizons within these
+            horizons categories.
+
+        zones: A Python dictionary where each key corresponds to the name of
+            the zones category where zone data need to be made empty. The value
+            associated to this key should be a list of zone names to modify. If
+            a string ``all`` is assigned instead of a list, all available zone
+            names for this category will be used.
+            Alternatively, if a list of zones categories is given instead of a
+            dictionary, the method will apply to all zones within these zones
+            categories.
+
+        grid_models: A Python dictionary where each key corresponds to the name
+            of the grid models where properties need to be made empty. The
+            value associated to this key should be a list of property names to
+            modify. If a string ``all`` is assigned instead of a list, all
+            available properties for this grid model name will be used.
+            Alternatively, if a list of grid models names is given instead of a
+            dictionary, the method will apply to all properties within these
+            grid models.
+    """
+
+    if not isinstance(config, dict):
+        raise TypeError("Argument must be a Python dictionary!")
+    assert "project" in config.keys(), "Input dict must contain key 'project'!"
+    project = config["project"]
+    assert "value" in config.keys(), "Input dict must contain key 'value'!"
+    value = config["value"]
+
+    def set_safe_empty(
+        project: _roxar.Project, surf_type: str, name: str, data_type: str
+    ):
+        """Set empty the horizon or zone surface.
+
+        Args:
+            project: the roxar project.
+            surf_type: the type of surface.
+                It should be the string "horizons" or "zones".
+            name: the name the of surface.
+        """
+        try:
+            if surf_type == "horizons":
+                surf = project.horizons[name][data_type]
+            else:
+                surf = project.zones[name][data_type]
+            surf.set_empty()
+            print(" >> >> " + name)
+        except Exception as e:
+            print(" >> >> " + name + " cannot be modified")
+            print(e)
+
+    def set_surfaces(
+        project: _roxar.Project, surf_type: str, dict_val: Union[List, Dict]
```

Comment:
Move this function outside the `set_data_empty` function?

Reply:
We could do that indeed, but do we want users to be able to use those functions independently ?